.github/workflows/ci.yml

```
@@ -45,8 +45,22 @@ jobs:
           sudo cp unicode.mapping /etc/apache2/
           sudo mkdir -p /var/cache/modsecurity
           sudo chown -R www-data:www-data /var/cache/modsecurity
+      - name: first check config (to get syntax errors)
+        run: sudo apachectl configtest
       - name: start apache with module
+        run: sudo systemctl restart apache2.service
+      - name: Search for errors/warnings in error log
         run: |
-          sudo systemctl restart apache2.service
-          sudo cat /var/log/apache2/error.log
-
+          # '|| :' handles the case grep doesn't match, otherwise the script exits with 1 (error)
+          errors=$(grep -E ':(?error|warn)[]]' /var/log/apache2/error.log) || :
+          if [[ -z "${errors}" ]]; then exit 0; fi
+          echo "Found errors/warnings in error.log"
```

Use GitHub commands for improved visibility in the log:
```suggestion
          echo "::error::Found errors/warnings in error.log"
```

---

Is "::error::" a kind of standard used somewhere else in the project?

---

I think the correct form is `::error`, without trailing `::` - see [this](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions) article. In CRS we use [this](https://github.com/coreruleset/coreruleset/blob/main/util/crs-rules-check/rules-check.py#L789) syntax too.

But I think @theseion will correct this.

---

I fixed it

---

Nope, the correct syntax is `::error::`. As you can see [here](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-error-message), the message follows after the second `::`. It's likely that GitHub is forgiving with how it parses the message, which is why the example airween posted above works.

---

It's not a "standard", but a way to format log output on GitHub.

---

Oh, so it's like
```
::error arg1=val1,arg2=val2,...argN=valN::
```
the trailing `::` is after the argument list of `error` "tag"?

---

Correct.